services/QuillLMS/engines/evidence/lib/evidence/evidence/check/plagiarism.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Evidence
+  class Check::Plagiarism < Check::Base
+
+    def run
+      rule = prompt.rules&.find_by(rule_type: Evidence::Rule::TYPE_PLAGIARISM)
+      feedback = get_plagiarism_feedback_from_previous_feedback(previous_feedback, rule)
+
+      if rule.plagiarism_texts.none?
+        @response = Evidence::PlagiarismCheck.new(entry, '', feedback, rule).feedback_object
+      else
+        plagiarism_check = nil
+        rule.plagiarism_texts.each do |plagiarism_text|
+          plagiarism_check = Evidence::PlagiarismCheck.new(entry, plagiarism_text.text, feedback, rule)
+          break unless plagiarism_check.feedback_object[:optimal]
+        end
+
+        @response = plagiarism_check.feedback_object
+      end
+    end
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    private def get_plagiarism_feedback_from_previous_feedback(prev, rule)
+      previous_plagiarism = prev.select {|f| f["feedback_type"] == Evidence::Rule::TYPE_PLAGIARISM && f["optimal"] == false }
+      feedbacks = rule&.feedbacks
+      previous_plagiarism.empty? ? feedbacks&.find_by(order: 0)&.text : feedbacks&.find_by(order: 1)&.text
+    end
```

Comment:
So, I'll add this to the description, but I think I could have been more clear on where some of this code came from:

For this PR, I copied the existing code from each of the check endpoints. Then I copied the tests from each endpoint to a `post :create` test and changed the constant in the `create` endpoint to only do that individual check. (To gain confidence that it's a direct port).

I don't disagree that there might be a refactor here, and I'd love feedback on this, but I think I'd rather refactor the endpoints in a separate PR, rather than do this move and refactor all at once.